src/com/palmergames/bukkit/towny/command/NationCommand.java

```diff
@@ -1735,10 +1735,12 @@ private void nationAlly(Player player, String[] split) throws TownyException {
 					ally = townyUniverse.getDataSource().getNation(name);
 					if (nation.equals(ally)) {
 						TownyMessaging.sendErrorMsg(player, TownySettings.getLangString("msg_own_nation_disallow"));
-						return;
+					} else if (nation.isAlliedWith(ally)) {
+						TownyMessaging.sendErrorMsg(player, String.format(TownySettings.getLangString("msg_already_ally"), ally));

```

Comment:
It's debatable whether the return statement should be put back in for the nation.equals(ally) and also added to your new if statement. I know it's supposed to pass over any non-usable names because it is sent a String[], but if you look at the cases below inside of the catch, if it is given an unusable argument it kills it with a return, rather than continuing to process the string. I would say put the returns in the try to match until all of the returns can be dealt with.

Reply:
The reason I removed the returns, is because it doesn't seem logical to have them when you think about it from the perspective of the player. For example

Nation1 is allied with Nation2
Nation2 is allied with Nation3

Player in Nation1 executes command `/n ally add Nation2 Nation3` 
Towny says you're already an ally and stops.

Now lets switch the order
Player in Nation1 executes command `/n ally add Nation3 Nation2`
Towny states that Nation1 is now an ally of Nation2
Towny then states Nation1 is already an ally with nation two.

^ There's a discrepancy in the results of these commands only when the order differs, which is confusing for the player. This discrepancy is caused by the order of the input.

By removing the `return` it removes this discrepancy, and simply "skips" and notifies if a nation in the input list if it's already allied, and then carries on to ally the other ones passed in. Therefore you can pass the nations into the command in any order and it should work as expected.